apps/endatix-hub/components/survey-creator.tsx

```diff
@@ -0,0 +1,171 @@
+        console.log(JSON.stringify(newCreator?.JSON));
+        callback(no, true);
+        };
+        setCreator(newCreator);
+    }
+
+  }, [creator, formJson, options]);
+
+  useEffect(() => {
+    if (isEditingName) {
+      document.addEventListener('mousedown', handleClickOutside);
+    } else {
+      document.removeEventListener('mousedown', handleClickOutside);
+    }
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside); // Clean up event listener
+    };
+  }, [isEditingName]);
+
+  const handleSaveAndGoBack = async () => {
+    try {
+      setIsSaving(true);
+
+      const updatedFormJson = creator?.JSON;
+
+      const result = await updateFormDefinitionJsonAction(formId, updatedFormJson);
+      if (!result.success) {
+        throw new Error(result.error);
+      }
+
+      router.push('/forms');
+    } catch (error) {
+      console.error('Failed to save form', error);
+    } finally {
+      setIsSaving(false);
+    }
+  };
+
+  const handleNameSave = async () => {
+    if (name !== originalName) {
+      startTransition(async () => {
+        await updateFormNameAction(formId, name);
+
+        setOriginalName(name);
+        setName(name);
+      });
+    }
+    setIsEditingName(false);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      handleNameSave();
+    } else if (e.key === 'Escape') {
+      setName(originalName);
+      setIsEditingName(false);
+    }
+  };
+
+  const handleClickOutside = (e: MouseEvent) => {
+    if (inputRef.current && !inputRef.current.contains(e.target as Node)) {
+      console.log("Clicked outside, waiting to save name...");
+      setTimeout(() => {
+        handleNameSave();
+      }, 0);
+    }
+  };
+
+  const toggleEnabled = async (enabled : boolean) => {
+    startTransition(async () => {
+        await updateFormStatusAction(formId, enabled);
+    })
+  }
+
+  return (
+    <div>
+        <div className="flex items-center mb-4 w-full">
+        <button
+            onClick={handleSaveAndGoBack}
+            className="mr-4 text-3xl flex items-center"
+            disabled={isSaving}
+            style={{ border: "none", background: "transparent" }}
+        >
+            ←
+        </button>
+
+        {isEditingName ? (
+          <input
+            ref={inputRef}
+            value={name}
+            onChange={(e) => setName(e.target.value)} // Update the name when typing
+            onKeyDown={handleKeyDown} // Handle Enter and Esc key presses
+            className="font-bold text-lg mr-16 border border-gray-300 rounded"
+            autoFocus
+          />
+        ) : (
+          <span
+            className="font-bold text-lg mr-16 hover:border hover:border-gray-300 hover:rounded px-1"
+            onClick={() => setIsEditingName(true)} // Click to enable editing
+            style={{ cursor: 'text' }} // Change cursor to I-beam (text editor cursor)
```

Comment:
Just a note. Since we use Tailwind, you can use `cursor-text` class instead of style - https://tailwindcss.com/docs/cursor